### DISCUSSION TITLE
Custom plugins using Tailwind CSS Play CDN

### DISCUSSION THREAD

**Discussion by liamwalder** [tailwindlabs/tailwindcss#8460] [Help] [answered]
Hey everyone,

So I am currently using BeyondCode Tailwind JIT compiler in my project, which has been working great for me. However, the time has come to move to Tailwind CSS V3 and therefore a migration to https://tailwindcss.com/docs/installation/play-cdn.

But I am running into one problem. Using the BeyondCode compiler, which I believe is V2 only, I am using custom plugins and as one example of this, I have a current one which looks like:

```
plugin(({ addVariant, e }) => {
    addVariant('current', ({ modifySelectors, separator }) => {
         modifySelectors(({ className, selector }) => {
            const isPseudoSelector = selector.endsWith(':before') || selector.endsWith(':after')
            const pseudo = isPseudoSelector ? `:${selector.split(':').pop()}` : ''
            return `.${e(`current${separator}${className}`)}[current]${pseudo}`
        })
    })
}),
```

However, I am struggling to migrate these plugins to be used in the new Tailwind CSS V3 Play CDN. So I guess I have two questions here.

The first being, am I able to use custom plugins in the current Play CDN version?

The second being, could someone point me in the right direction on how I would do this? I have been pointed in the direction of https://tailwindcss.com/docs/plugins#adding-variants before but still struggled to get it working. But of course, I am happy to revisit and have more of a play with this if this is the correct way.

Thanks in advance for your help. This is an awesome project which I rely one heavily, Tailwind CSS in general.

Thanks,
Liam

**Comment by liamwalder** [comment, marked as answer]
Interacted with @adamwathan on Twitter with this response.

<img width="587" alt="Screenshot 2022-05-30 at 17 27 26" src="https://user-images.githubusercontent.com/5079748/171136964-83e05c59-a8bf-4c2a-8f3f-8825f7b7ba82.png">

**Reply by g-danna** [reply]
Thanks for sharing here! 🙏 Here's an alt text for those who need.

Adam Wathan replied: 
> "Should be able to do this! Just build a real CSS file for real projects though, this whole project is not meant to actually be used, it's just a toy."

The attached code:
```html
<!doctype html>
<html>
   <head>
      <meta charset="UTF-8">
      <meta name= "viewport" content="width=device-width, initial-scale=1.0">
      <script src="https://cdn.tailwindcss.com"></script>
      <script>
        tailwind.config = {
          //...
          plugins: [
             function ({ addVariant }) {
                addVariant ('poop', ' .poop &')
             }
          ]
        }
      </script> 
   </head>
   <body>
      <h1 class="text-3xl font-bold underline **text-clifford**">
         Hello world!
      </h1>
   </body>
</html>
```

**Comment by tresorama** [comment]
Is currently usable ?

**Reply by tresorama** [reply]
Replying to myself. 
It works.
